packages/cli/src/commands/container/release.ts

```diff
@@ -0,0 +1,111 @@
+import color from '@heroku-cli/color'
+import {Command, flags} from '@heroku-cli/command'
+import {ux} from '@oclif/core'
+import * as Heroku from '@heroku-cli/schema'
+import {debug} from '../../lib/container/debug'
+import {streamer} from '../../lib/container/streamer'
+
+type ImageResponse = {
+  schemaVersion: number,
+  history: [{
+    v1Compatibility: string,
+    }
+  ],
+  config: {
+    digest: string
+  }
+}
+
+export default class Release extends Command {
+  static topic = 'container'
+  static description = 'Releases previously pushed Docker images to your Heroku app'
+  static usage = 'heroku container:release'
+  static example = `
+  ${color.cmd('heroku container:release web')}        # Releases the previously pushed web process type
+  ${color.cmd('heroku container:release web worker')} # Releases the previously pushed web and worker process types`
+
+  static strict = false
+
+  static flags = {
+    app: flags.app({required: true}),
+    verbose: flags.boolean({char: 'v'}),
+  }
+
+  async run() {
+    const {flags, argv, args} = await this.parse(Release)
+    const {app, verbose} = flags
+
+    if (argv.length === 0) {
+      this.error(`Error: Requires one or more process types\n ${Release.example}`)
+    }
+
+    if (verbose) {
+      debug.enabled = true
+    }
+
+    await this.heroku.get(`/apps/${app}`)
+
+    const herokuHost: string = process.env.HEROKU_HOST || 'heroku.com'
+    const updateData: any[] = []
+    for (const process of argv) {
+      const image = `${app}/${process}`
+      const tag = 'latest'
+      const {body: imageResp} = await this.heroku.get<ImageResponse>(
+        `/v2/${image}/manifests/${tag}`,
+        {
+          hostname: `registry.${herokuHost}`,
+          headers: {
+            Accept: 'application/vnd.docker.distribution.manifest.v2+json',
+          }},
```

Comment:
The authentication issue you're facing has to do with the Heroku Registry only accepting `Basic Auth` as authentication method, whereas the `APIClient` in use authenticates using a `Bearer Token` scheme. Maybe our current CLI uses basic auth to authenticate against API too and thus doesn't fail.

In order to fix the problem we need to force the `Authorization` header here as follows:

```suggestion
            Authorization: `Basic ${Buffer.from(`:${this.heroku.auth}`).toString('base64')}`,
          }},
```

Reply:
This worked perfectly, thank you Santi!!